corehq/apps/export/models/new.py

```diff
@@ -0,0 +1,191 @@
+    last_occurrence = IntegerProperty()
+
+
+class ExportColumn(DocumentSchema):
+    item = SchemaProperty(ExportItem)
+    name = StringProperty()
+    path = StringProperty()
+    label = StringProperty()
+    last_occurrence = IntegerProperty()
+
+    def get_value(self, doc, base_path):
+        """
+        Get the value of self.item of the given doc.
+        When base_path is [], doc is a form submission or case,
+        when base_path is non empty, doc is a repeat group from a form submission.
+        :param doc: A form submission or instance of a repeat group in a submission or case
+        :param base_path:
+        :return:
+        """
+        # Confirm the ExportItem's path starts with the base_path
+        assert base_path == self.item.path[:len(base_path)]
+        # Get the path from the doc root to the desired ExportItem
+        path = self.item.path[len(base_path):]
+        return NestedDictGetter(path)(doc)
+
+
+class TableConfiguration(DocumentSchema):
+
+    name = StringProperty()
+    repeat_path = StringProperty()
+    columns = ListProperty(ExportColumn)
+
+    def get_rows(self, document):
+        """
+        Return a list of ExportRows generated for the given document.
+        :param document: dictionary representation of a form submission or case
+        :return: List of ExportRows
+        """
+        # Note that repeat_items will be [document] if self.repeat_path is []
+        repeat_items = self._get_items_for_repeat(self.repeat_path, [document])
+        rows = []
+        for item in repeat_items:
+            row = ExportRow(data=[
+                col.get_value(item, self.repeat_path) for col in self.columns
+            ])
+            rows.append(row)
+        return rows
+
+    def _get_items_for_repeat(self, path, docs):
+        """
+        Return each instance of a repeat group at the path from the given docs.
+        If path is [], just return the docs
+
+        >>> TableConfiguration()._get_items_for_repeat(['foo'], [{'foo': {'bar': 'a'}}, {'foo': {'bar': 'b'}}])
+        [{'bar': 'a'}, {'bar': 'b'}]
+        >>> TableConfiguration()._get_items_for_repeat(['foo', 'bar'], [{'foo': [{'bar': {'baz': 'a'}}, {'bar': {'baz': 'b'}},]}]
+        [{'baz': 'a'}, {'baz': 'b'}]
+
+        :param path: A list of a strings
+        :param docs: A list of dicts representing form submissions
+        :return:
+        """
+        if len(path) == 0:
+            return docs
+
+        new_docs = []
+        for doc in docs:
+            next_doc = doc.get(path[0], {})
+            if type(next_doc) == list:
+                new_docs.extend(next_doc)
+            else:
+                new_docs.append(next_doc)
+        return self._get_items_for_repeat(path[1:], new_docs)
+
+
+class FormExportConfiguration(Document):
+
+    class Meta:
+        app_label = 'export'
+
+
+class ExportRow(object):
+    def __init__(self, data):
+        self.data = data
+
+
+class RepeatGroup(DocumentSchema):
```

Comment:
we can probably drop this. `get_questions` returns a flat list and everything we need doesn't preserve the hierarchy. i think it'd be enough to have `data_type=repeat`